### DISCUSSION TITLE
ListBox with multiple DataTemplate

### DISCUSSION THREAD

**Discussion by JohnKesko** [AvaloniaUI/Avalonia#12984] [General]
Can we have multiple DataTemplate inside a ListBox? If I remember, there was a thing in WPF that we could do, like: DataTemplateSelector or something but can we do it in Avalonia as well?
	

 
        <ListBox Name="AlbumListBox" Grid.Row="1" Grid.Column="2" 
                 x:DataType="local:MainWindow" 
                 ItemsSource="{Binding AlbumItems}" 
                 SelectionMode="Single" 
                 SelectionChanged="AlbumListBox_OnSelectionChanged">
            
            
            <ListBox.ItemsPanel>
                <ItemsPanelTemplate>
                    <UniformGrid Rows="3" Columns="3"/>
                </ItemsPanelTemplate>
            </ListBox.ItemsPanel>

            
            <ListBox.ItemTemplate>
                <DataTemplate x:DataType="{x:Type dto:Photo}" >
                    <Border Margin="5" BorderBrush="Black" BorderThickness="1">
                        <Image Source="{Binding ImageBitmap}" Width="100" Height="100"/>
                    </Border>
                </DataTemplate>
            
                <DataTemplate x:DataType="{x:Type dto:Video}">
                    
                    <StackPanel Orientation="Horizontal">
                        <TextBlock Text="🎥" FontSize="24"/>
                        <TextBlock Text="{Binding Name}" Margin="5,0,0,0"/>
                    </StackPanel>
                </DataTemplate>
            </ListBox.ItemTemplate>
        </ListBox>

**Comment by wieslawsoltes** [comment]
<ListBox.DataTemplates>
...
</ListBox.DataTemplates>

**Reply by JohnKesko** [reply]
Look at that, it works :) Thanks!

**Comment by timunie** [comment]
you can use `FunkDataTemplate` or implement `IDataTemplate` which is way more powerful than WPFs DataTemplateSelector. Check out: https://github.com/AvaloniaUI/Avalonia.Samples#%EF%B8%8F-datatemplate-samples

**Reply by JohnKesko** [reply]
Thanks! Will check that out as well.

**Comment by JohnKesko** [comment]
@wieslawsoltes Solution works great, closing.
